R/functions.R

```diff
@@ -25,15 +25,23 @@ get_data <- function(synID, version = NULL){
 #'@param factors A vector of factor variables
 #'@param continuous A vector of continuous variables
 #' Remove primary variable for now - A vector of primary variables
-#'@param sample_variable A character vector with the column name corresponding to sample identifiers
 #'
 #'@export
 #'@return A data frame with coerced variables.
 #'@examples
-clean_covariates <- function(md, factors, continuous, sample_variable){
-  md[,factors] <- lapply(md[,factors], factor)
-  md[,continuous] <- lapply(md[,continuous], as.numeric)
-  md
+clean_covariates <- function(md, factors, continuous){
+  if (missing(factors) | missing(continuous)) {
+    stop("Factor and continuous variables are required.")
+  } else if (length(intersect(factors, continuous)) != 0) {
+    stop("Variables are present in both the continuous and factor arguments. Variables must be designated
+         numeric or factor, not both.")
+  } else if (!(factors %in% colnames(md) | !(continuous %in% colnames(md)))) {
```

Comment:
Are there ever cases where all the columns in the data are numeric or all of them are factors? This line will fail if there's not at least one column of each.

Reply:
There certainly could be cases where a user might want to only specify continuous variables OR factor variables. Tracking this issue in #10.